admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/service/impl/SyncAuthTokenServiceImpl.java

```diff
@@ -124,6 +131,9 @@ public String getAuthToken(String requestData) {
             Machine machine = validateRequestData(header, payload, signature);
             try {
                 MachineAuthDto machineAuthDto = objectMapper.readValue(payload, MachineAuthDto.class);
+                if (machineAuthDto.getMachineName() != null && !machineAuthDto.getMachineName().equalsIgnoreCase(machine.getId())) {
+                	throw new RuntimeException();
```

Comment:
Why is it RuntimeException ? 
lets throw request exception with specific error code.